Foody/Onboarding/Register/RegisterViewController.swift

```diff
@@ -6,24 +6,161 @@
+        datePicker.frame.size = CGSize(width: 0, height: 300)
+        datePicker.preferredDatePickerStyle = .wheels
+        
+        datePicker.maximumDate = Date()
+        
+        createDatePicker()
     }
     
-
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+    @objc func dateChange(datePicker: UIDatePicker) {
+        dateText.text = formatDate(date: datePicker.date)
+    }
+    
+    func createDatePicker() {
+        
+      
+        let toolbar = UIToolbar()
+        toolbar.sizeToFit()
+        
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: nil, action: #selector(doneTapped))
+        toolbar.setItems([doneButton], animated: true)
+        dateText.inputAccessoryView = toolbar
+        
+        dateText.inputView = datePicker
+        datePicker.datePickerMode = .date
+    }
+    @objc func doneTapped () {
+        
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .none
+        
+        dateText.text = formatter.string(from: datePicker.date)
+        self.view.endEditing(true)
+    }
+    
+    func formatDate(date: Date) -> String {
+        
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd/MM/yyyy"
+        return formatter.string(from: date)
+    }
+    
+    func dateDayCount(date:String,format:String = "dd/MM/yyyy") -> Int {
+        
+        let dateFormatter = DateFormatter()
+        
+        dateFormatter.locale = Locale.current
+        dateFormatter.timeZone = TimeZone.autoupdatingCurrent
+        dateFormatter.timeStyle = .short
+        dateFormatter.dateFormat = format
+        
+        let calender = Calendar.current
+        let firstDate = calender.startOfDay(for: dateFormatter.date(from: date)!)
+        let twiceDate = calender.startOfDay(for: Date())
+        let companents = calender.dateComponents([.year], from: firstDate, to: twiceDate)
+        
+        if companents.year! >= 15 {
+            
+            makeAlert(title: "Registration Successful.", message:"You are welcome. :)" )
+            
+            //Navigation
+            
+        }else {
+            makeAlert(title: "Warning!", message: "Persons under the age of 15 cannot enter.")
+        }
+        return companents.year!
+    }
+    
+    @IBAction func signUpClicked(_ sender: Any) {
+        
+        // email and password validation
+        
+        if let username = usernameText.text , let email = emailText.text , let password = passwordText.text {
+            if username == "" && email == "" && password == "" {
```

Comment:
Instead of username == "" use username.isEmpty, you can apply same for other parameters.